File: salasar-art-craft/fix_encoding.py
import os
import re

def process_dir(directory):
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith('.tsx') or file.endswith('.ts'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                # Replace the weird 'â‚¹' or ',1' right before {*.toLocaleString('en-IN')} with '₹'
                new_content = re.sub(
                    r'[^\x00-\x7F]+\{([A-Za-z0-9_\.]+\.toLocaleString\(\'en-IN\'\))\}',
                    r'₹{\1}',
                    content
                )
                
                # Also replace any literal 'â‚¹' characters
                new_content = new_content.replace('â‚¹', '₹')
                
                # Also fix the `?` and `,1` variants if any
                new_content = re.sub(
                    r'(\?|,1)\{([A-Za-z0-9_\.]+\.toLocaleString\(\'en-IN\'\))\}',
                    r'₹{\2}',
                    new_content
                )

                if new_content != content:
                    with open(path, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Fixed {path}")

process_dir('src')

File: salasar-art-craft/fix_encoding.js
const fs = require('fs');
const path = require('path');

function walkSync(dir, filelist = []) {
  fs.readdirSync(dir).forEach(file => {
    const dirFile = path.join(dir, file);
    if (fs.statSync(dirFile).isDirectory()) {
      filelist = walkSync(dirFile, filelist);
    } else {
      if (dirFile.endsWith('.tsx') || dirFile.endsWith('.ts')) {
        filelist.push(dirFile);
      }
    }
  });
  return filelist;
}

const files = walkSync('./src');
files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  
  // Replace the corrupted characters
  let changed = false;
  
  if (content.includes('â‚¹')) {
    content = content.replace(/â‚¹/g, '₹');
    changed = true;
  }
  
  if (content.match(/[^\x00-\x7F]*\{.*toLocaleString\('en-IN'\)\}/)) {
    // Specifically target the exact symbol right before {something.toLocaleString('en-IN')}
    content = content.replace(/([^\x00-\x7F]+|,1|\?|\,1)\{([A-Za-z0-9_\.]+\.toLocaleString\('en-IN'\))\}/g, '₹{}');
    changed = true;
  }

  // specifically catch ,1 as it was printed in powershell, might just be â‚¹ in UTF-8 buffer but we can just use the regex
  content = content.replace(/,1/g, '₹');

  if (changed) {
    fs.writeFileSync(file, content, 'utf8');
  }
});
console.log('Fixed encoding issues in all TSX files.');
